website/docs/language/meta-arguments/resource-provider.mdx

```diff
@@ -47,12 +47,54 @@ resource "google_compute_instance" "example" {
 }
 ```
 
-A resource always has an implicit dependency on its associated provider, to
-ensure that the provider is fully configured before any resource actions
-are taken.
+If you select a provider configuration that uses `for_each` then you
+must also dynamically select a different instance of the provider configuration for
+each instance of the resource by including an instance key expression in brackets:
 
-The `provider` meta-argument expects
-[a `<PROVIDER>.<ALIAS>` reference](../../language/providers/configuration.mdx#referring-to-alternate-provider-configurations),
-which does not need to be quoted. Arbitrary expressions are not permitted for
-`provider` because it must be resolved while OpenTofu is constructing the
-dependency graph, before it is safe to evaluate expressions.
+```hcl
+variable "aws_regions" {
+  type = map(object({
+    vpc_cidr_block = string
+  }))
+}
+
+provider "aws" {
+  alias    = "by_region"
+  for_each = var.aws_regions
+
+  region = each.key
+}
+
+resource "aws_vpc" "private" {
+  # This expression filters var.aws_regions to include only
+  # the elements whose value is not null. Refer to the
```

Comment:
We could probably move this example to a file and embed the identical contents in both locations

Reply:
Is that something our website platform supports? I'm not super familiar with it, but if you have an example from elsewhere that I can learn from then I'm happy to learn from it. :grinning:

Reply:
In a conversation elsewhere we found this example, which I've taken inspiration from:

https://github.com/opentofu/opentofu/blob/f6a1d8e2dd6505c2f60c39955a10c473ed2210a5/website/docs/intro/install/standalone.mdx#L11-L17

https://github.com/opentofu/opentofu/blob/f6a1d8e2dd6505c2f60c39955a10c473ed2210a5/website/docs/intro/install/standalone.mdx#L55-L65